sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala

```diff
@@ -44,7 +44,7 @@ case class UnresolvedNamespace(multipartIdentifier: Seq[String]) extends Unresol
 case class UnresolvedTable(
     multipartIdentifier: Seq[String],
     commandName: String,
-    relationTypeMismatchHint: Option[String]) extends UnresolvedLeafNode
+    hint: Boolean = false) extends UnresolvedLeafNode
```

Comment:
This is not a good name, as `hint` is too vague. How about `suggestAlterCommand: Boolean`?

Reply:
Okay, let me refactor it.